backend/compact-connect/lambdas/nodejs/ingest-event-reporter/lib/event-client.ts

```diff
@@ -0,0 +1,164 @@
+    public getLastWeekTimestamps() {
+        const today: Date = new Date();
+        const lastWeek: Date = new Date();
+
+        today.setUTCHours(0, 0, 0, 0);
+        lastWeek.setUTCHours(0, 0, 0, 0);
+
+        lastWeek.setUTCDate(today.getUTCDate() - 7);
+        // Uncomment to manually force today's events into the time window (for development/testing)
+        // today.setUTCDate(today.getUTCDate() + 1);
+
+        return [
+            Number.parseInt(
+                (lastWeek.valueOf()/1000).toString()
+            ),
+            Number.parseInt(
+                (today.valueOf()/1000).toString()
+            )
+        ];
+    }
+
+    /*
+     * Queries the data event table for validation errors by looking for each comp
+     */
+    public async getValidationErrors(
+        compact: string,
+        jurisdiction: string,
+        startTimeStamp: number,
+        endTimeStamp: number
+    ): Promise<IValidationErrorEventRecord[]> {
+        this.logger.info('Getting validation errors', {
+            compact: compact,
+            jurisdiction: jurisdiction,
+            start_time_stamp: startTimeStamp,
+            end_time_stamp: endTimeStamp,
+        });
+        const tableName = environmentVariables.getDataEventTableName();
+        const resp = await this.dynamoDBClient.send(new QueryCommand({
+            TableName: tableName,
+            Select: 'ALL_ATTRIBUTES',
+            // We don't really want to present more than 100 errors in an email, so we won't bother
+            // querying that many up
+            Limit: 100,
+            KeyConditionExpression: 'pk = :pk and sk BETWEEN :skBegin and :skEnd',
+            ExpressionAttributeValues: {
+                ':pk': { 'S': `COMPACT#${compact}#JURISDICTION#${jurisdiction}` },
+                ':skBegin': { 'S': `TYPE#license.validation-error#TIME#${startTimeStamp}#` },
+                ':skEnd': { 'S': `TYPE#license.validation-error#TIME#${endTimeStamp}#` },
+            }
+        }));
+
+        return resp.Items?.map((item) => unmarshall(item) as IValidationErrorEventRecord) || [];
+    }
+
+    /*
+     * Queries the data event table for ingest failures by looking for each comp
+     */
+    public async getIngestFailures(
+        compact: string,
+        jurisdiction: string,
+        startTimeStamp: number,
+        endTimeStamp: number
+    ): Promise<IIngestFailureEventRecord[]> {
+        this.logger.info('Getting ingest failures', {
+            compact: compact,
+            jurisdiction: jurisdiction,
+            start_time_stamp: startTimeStamp,
+            end_time_stamp: endTimeStamp,
+        });
+        const resp = await this.dynamoDBClient.send(new QueryCommand({
+            TableName: environmentVariables.getDataEventTableName(),
+            Select: 'ALL_ATTRIBUTES',
+            // We don't really want to present more than 100 errors in an email, so we won't bother
+            // querying that many up
+            Limit: 100,
```

Comment:
My current assumption (hope) is that, if there are more than 100 errors in a single day, it's because the state has some consistent error across all/most of their licenses, probably while they are just getting integrated. In that case, displaying the same error up to 30,000 times, is probably not helpful. Instead, we'll report the first 100, let them fix those and try again.

If it turns out that we really need to support reporting back a higher volume of errors, we'll have to evaluate some enhancement that is probably not sending a report out via email.